README.md

```diff
@@ -0,0 +1,219 @@
+    $ cd bar
+
+And configure the dependency, note the `repo` configuration option.
+
+```elixir
+  defp deps() do
+    [
+      {:foo, "~> 0.1", repo: "test_repo"}
+    ]
+  end
+```
+
+Since the development server includes a `hexpm_mirror` repo, let's try that too:
+
+    $ HEX_MIRROR_URL=http://localhost:4000/repos/hexpm_mirror mix hex.package fetch decimal 1.8.0
+
+See [`mix help hex.config`](https://hexdocs.pm/hex/Mix.Tasks.Hex.Config.html) for more information
+about configuring your Hex installation.
+
+### Usage with Rebar3
+
+Let's create a new package and publish it to our `test_repo` repository:
+
+    $ rebar3 new lib baz
+    $ cd baz
+
+Now, let's configure Rebar3 to use our custom repo:
+
+```erlang
+%% ~/.config/rebar3/rebar.config
+{plugins, [rebar3_hex]}.
+{hex, [
+  {repos, [
+    #{
+      name => <<"test_repo">>,
+
+      %% client requires an API key, but since NimbleHex does not
+      %% support authentication, you can put any key here
+      api_key => <<"does-not-matter">>,
+
+      api_url => <<"http://localhost:4000/api">>,
+      api_repository => <<"test_repo">>
+    }
+  ]}
+]}.
+```
+
+And publish the package specifying the repo:
+
+    $ rebar3 hex publish -r test_repo
+
+**TODO** Finally, let's use this package from another project:
+
+### Deployment with releases
+
+It's recommended to deploy NimbleHex with Elixir releases.
+
+Let's now assemble the release locally:
+
+    $ MIX_ENV=prod mix release
+    * assembling nimble_hex-0.1.0 on MIX_ENV=prod
+    (...)
+
+And start it:
+
+    PORT=4000 \
+    NIMBLE_HEX_URL=http://localhost:4000 \
+    _build/prod/rel/nimble_hex/bin/nimble_hex start
+
+As you can see, some configuration can be set by adjusting system environment variables,
+see [`config/releases.exs`](config/releases.exs)
+
+Also, see [`mix help release`](https://hexdocs.pm/mix/Mix.Tasks.Release.html?) for general
+information on Elixir releases.
+
+## More information
+
+See following modules documentation to learn more about given feature:
+
+* [`NimbleHex.Store.Local`](lib/nimble_hex/store/local.ex)
+
+* [`NimbleHex.Store.S3`](lib/nimble_hex/store/s3.ex)
+
+* [`NimbleHex.Mirror`](lib/nimble_hex/mirror.ex)
+
+## Contributing
+
+The goal of NimbleHex is to provide a minimal server that implements Hex specifications. Why
+minimal? By keeping the project focused on bare minimum we hope it's easy to understand and
+serves as a good starting point for a more complete solution that makes sense in a given
+organization. A production grade system should include infrastructure for monitoring, backups,
+SSL, and more, not to mention features like user management, SSO and such, but that's out of
+the scope of NimbleHex project.
+
+We welcome anyone to contribute to the project, especially around documentation and guides,
+but features specific to narrow set of users likely won't be accepted. For a complete and
+production-grade implementation see source code of [Hex.pm](https://github.com/hexpm/hexpm).
```

Comment:
Should we clarify further that hexpm is not for running your organization's Hex server, rather it's for running Hex.pm specifically which has very different constraints?

Reply:
good call. How about:

For a complete and
 production-grade implementation see source code of [Hex.pm](https://github.com/hexpm/hexpm), however keep in mind it's optimised for running the public registry for the community and thus have different features and constraints than a self-hosted solution might have.

Reply:
Great!